src/tests/class_tests/openms/source/PeptideIdentificationList_test.cpp

```diff
@@ -0,0 +1,293 @@
+  pep_id2.setRT(300.0);
+  pep_id2.setMZ(400.0);
+  
+  PeptideIdentificationList pep_ids{pep_id1, pep_id2};
+  TEST_EQUAL(pep_ids.size(), 2)
+  TEST_REAL_SIMILAR(pep_ids[0].getRT(), 100.0)
+  TEST_REAL_SIMILAR(pep_ids[0].getMZ(), 200.0)
+  TEST_REAL_SIMILAR(pep_ids[1].getRT(), 300.0)
+  TEST_REAL_SIMILAR(pep_ids[1].getMZ(), 400.0)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList(const PeptideIdentificationList&)))
+{
+  PeptideIdentificationList pep_ids1;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  pep_ids1.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids2(pep_ids1);
+  TEST_EQUAL(pep_ids2.size(), 1)
+  TEST_REAL_SIMILAR(pep_ids2[0].getRT(), 1234.5)
+  TEST_REAL_SIMILAR(pep_ids2[0].getMZ(), 567.8)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList(PeptideIdentificationList&&)))
+{
+  PeptideIdentificationList pep_ids1;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  pep_ids1.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids2(std::move(pep_ids1));
+  TEST_EQUAL(pep_ids2.size(), 1)
+  TEST_REAL_SIMILAR(pep_ids2[0].getRT(), 1234.5)
+  TEST_REAL_SIMILAR(pep_ids2[0].getMZ(), 567.8)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList& operator=(const PeptideIdentificationList&)))
+{
+  PeptideIdentificationList pep_ids1;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  pep_ids1.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids2;
+  pep_ids2 = pep_ids1;
+  TEST_EQUAL(pep_ids2.size(), 1)
+  TEST_REAL_SIMILAR(pep_ids2[0].getRT(), 1234.5)
+  TEST_REAL_SIMILAR(pep_ids2[0].getMZ(), 567.8)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList& operator=(PeptideIdentificationList&&)))
+{
+  PeptideIdentificationList pep_ids1;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  pep_ids1.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids2;
+  pep_ids2 = std::move(pep_ids1);
+  TEST_EQUAL(pep_ids2.size(), 1)
+  TEST_REAL_SIMILAR(pep_ids2[0].getRT(), 1234.5)
+  TEST_REAL_SIMILAR(pep_ids2[0].getMZ(), 567.8)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList& operator=(const std::vector<PeptideIdentification>& vec)))
+{
+  std::vector<PeptideIdentification> base_vector;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  base_vector.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids;
+  pep_ids = base_vector;
+  TEST_EQUAL(pep_ids.size(), 1)
+  TEST_REAL_SIMILAR(pep_ids[0].getRT(), 1234.5)
+  TEST_REAL_SIMILAR(pep_ids[0].getMZ(), 567.8)
+}
+END_SECTION
+
+START_SECTION((PeptideIdentificationList& operator=(std::vector<PeptideIdentification>&& vec)))
+{
+  std::vector<PeptideIdentification> base_vector;
+  PeptideIdentification pep_id;
+  pep_id.setRT(1234.5);
+  pep_id.setMZ(567.8);
+  base_vector.push_back(pep_id);
+  
+  PeptideIdentificationList pep_ids;
+  pep_ids = std::move(base_vector);
```

Comment:
ideally, check that capacity() of base_vector is `0` afterwards, i.e. internally this was a true move and not a move of elements or even a full copy.